airflow/providers/amazon/aws/hooks/quicksight.py

```diff
@@ -156,16 +156,17 @@ def wait_for_state(
         sec = 0
         status = self.get_status(aws_account_id, data_set_id, ingestion_id)
         while status in self.NON_TERMINAL_STATES and status != target_state:
+            time.sleep(check_interval)
             self.log.info("Current status is %s", status)
+            status = self.get_status(aws_account_id, data_set_id, ingestion_id)
             if status in self.FAILED_STATES:
                 info = self.get_error_info(aws_account_id, data_set_id, ingestion_id)
                 raise AirflowException(f"The Amazon QuickSight Ingestion failed. Error info: {info}")
             if status == "CANCELLED":
                 raise AirflowException("The Amazon QuickSight SPICE ingestion cancelled!")
```

Comment:
I would do it that way to avoid duplicate calls on `self.get_status`
```suggestion
        while True:
            status = self.get_status(aws_account_id, data_set_id, ingestion_id)
            self.log.info("Current status is %s", status)
            if status in self.FAILED_STATES:
                info = self.get_error_info(aws_account_id, data_set_id, ingestion_id)
                raise AirflowException(f"The Amazon QuickSight Ingestion failed. Error info: {info}")
            if status == "CANCELLED":
                raise AirflowException("The Amazon QuickSight SPICE ingestion cancelled!")
            if status not in self.NON_TERMINAL_STATES or status == target_state:
                break
            time.sleep(check_interval)
```

Reply:
Also, `sec` is not used so it is safe to remove it

Reply:
shall we initialise cancel rather than hardcoding it, like we have others states?  

`CANCELLED_STATES = {"CANCELLED"}`

Reply:
This is nice, simplifies the logic quite a bit. 

> shall we initialise cancel rather than hardcoding it, like we have others states?

It's only one item, but on the other hand `FAILED_STATES` is only one item and we have a set for it. So I'd +1 doing the same for cancelled for consistency sake.